applications/ecallmgr/src/ecallmgr_conference_command.erl

```diff
@@ -0,0 +1,258 @@
+    case kapi_conference:unlock_v(JObj) of
+        'false' ->
+            {'error', <<"conference unlock failed to execute as JObj did not validate.">>};
+        'true' ->
+            {<<"unlock">>, <<>>}
+    end;
+
+get_conf_command(<<"record">>, _Focus, _ConferenceId, JObj) ->
+    case kapi_conference:record_v(JObj) of
+        'false' ->
+            {'error', <<"conference record failed to execute as JObj did not validate.">>};
+        'true' ->
+            MediaName = kz_json:get_value(<<"Media-Name">>, JObj),
+            RecordingName = ecallmgr_util:recording_filename(MediaName),
+            {<<"recording">>, [<<"start ">>, RecordingName]}
+    end;
+
+get_conf_command(<<"recordstop">>, _Focus, _ConferenceId, JObj) ->
+    case kapi_conference:recordstop_v(JObj) of
+        'false' -> {'error', <<"conference recordstop failed validation">>};
+        'true' ->
+            MediaName = ecallmgr_util:recording_filename(kz_json:get_binary_value(<<"Media-Name">>, JObj)),
+            {<<"recording">>, [<<"stop ">>, MediaName]}
+    end;
+
+get_conf_command(<<"tones">>, _Focus, _ConferenceId, JObj) ->
+    case kapi_conference:tones_v(JObj) of
+        'false' -> {'error', <<"conference tones failed to validate">>};
+        'true' ->
+            Tones = kz_json:get_value(<<"Tones">>, JObj, []),
+            FSTones = [begin
+                           Vol = case kz_json:get_value(<<"Volume">>, Tone) of
+                                     'undefined' -> [];
+                                     %% need to map V (0-100) to FS values
+                                     V -> list_to_binary(["v=", kz_util:to_list(V), ";"])
+                                 end,
+                           Repeat = case kz_json:get_value(<<"Repeat">>, Tone) of
+                                        'undefined' -> [];
+                                        R -> list_to_binary(["l=", kz_util:to_list(R), ";"])
+                                    end,
+                           Freqs = string:join([ kz_util:to_list(V) || V <- kz_json:get_value(<<"Frequencies">>, Tone) ], ","),
+                           On = kz_util:to_list(kz_json:get_value(<<"Duration-ON">>, Tone)),
+                           Off = kz_util:to_list(kz_json:get_value(<<"Duration-OFF">>, Tone)),
+                           kz_util:to_list(list_to_binary([Vol, Repeat, "%(", On, ",", Off, ",", Freqs, ")"]))
+                       end || Tone <- Tones],
+            Arg = "tone_stream://" ++ string:join(FSTones, ";"),
+            {<<"play">>, Arg}
+    end;
+
+%% The following conference commands can optionally specify a participant
+get_conf_command(<<"play">>, _Focus, ConferenceId, JObj) ->
+    case kapi_conference:play_v(JObj) of
+        'false' ->
+            {'error', <<"conference play failed to execute as JObj did not validate.">>};
+        'true' ->
+            UUID = kz_json:get_ne_value(<<"Call-ID">>, JObj, ConferenceId),
```

Comment:
How about local functions to fetch this and Participant-ID too (or any fields fetched in more than one place)?
This way they would be handled with the same code.
Also, what about get_ne_binary_value in both cited examples?